chainstate/src/detail/spend_cache.rs

```diff
@@ -201,28 +235,40 @@ impl<'a> CachedInputs<'a> {
                             tx_id: Some(tx.get_id().into()),
                             source_output_index: outpoint.get_output_index() as usize,
                         })?;
-                    output.get_destination().clone()
+                    verify_signature(output.get_destination(), tx, input_idx)
+                        .map_err(|_| BlockError::SignatureVerificationFailed(tx.get_id()))?;
                 }
-                SpendablePosition::BlockReward(_reward_pos) => {
+                SpendablePosition::BlockReward(reward_pos) => {
                     // TODO(Roy): fill this with the block reward that the user now is spending
-                    todo!()
+
+                    let block = self
+                        .db_tx
+                        .get_block(reward_pos.clone())
```

Comment:
To get a block reward in history, you do not need a whole block. You only need the header. And the header can be found in the block index.

Reply:
Thanks!